app/views/unattended/partition_tables_templates/preseed_default.erb

```diff
@@ -29,6 +29,15 @@ d-i partman/early_command string \
 # The presently available methods are: "regular", "lvm" and "crypto"
 d-i partman-auto/method string <%= partitioning_method %>
 
+# Unattended 'crypto' partitioning method requires a passphrase.
+<% if partitioning_method == 'crypto' -%>
+d-i partman-crypto/passphrase password <%= host_param('partitioning-crypto-passphrase') %>
+d-i partman-crypto/passphrase-again password <%= host_param('partitioning-crypto-passphrase') %>
+
+# When disk encryption is enabled, skip wiping the partitions beforehand.
+d-i partman-auto-crypto/erase_disks boolean <%= host_param_true?('partitioning-crypto-erase-disks') ? 'true' : 'false' %>
+<% end -%>
+
```

Comment:
Nit: if you write it like this don't end up with an empty line on non-crypto:
```suggestion

<% end -%>
```

Reply:
If you look at a preview you may see many empty lines. This has to do with these kind of blocks. For example:
```erb
# Option 1
option=1

<% if some_condition -%>
# Something else
something=else
<% end -%>

# Closing option
close=true
```

When the condition is true, this renders to:
```
# Option 1
option=1

# Something else
something=else

# Closing option
close=true
```

This is great and what we would expect. However, when the condition is false we see:
```
# Option 1
option=1


# Closing option
close=true
```

With the style I suggested you will see
```
# Option 1
option=1

# Closing option
close=true
```

It's just a slight difference, but if you have a bunch of them it can add up.

Note that this same thing applies to configuration files (for example with Puppet). The effect is that even if the option defaults to false, the config file is rewritten and that typically results in a service restart. By using the style I suggested you can avoid that service restart, which minimizes downtime.

Reply:
OK, I understand the goal. But the suggested changes does produce extra lines too.

Reply:
It could be that the GH suggestion doesn't look quite ok. What I'm suggesting is (building on the previous example):
```erb
# Option 1
option=1

<% if some_condition -%>
# Something else
something=else

<% end -%>
# Closing option
close=true
```